operator/v1/types_csi_cluster_driver.go

```diff
@@ -118,25 +121,129 @@ const (
+
+	// gcp is used to configure the GCP CSI driver.
+	// +optional
+	GCP *GCPCSIDriverConfigSpec `json:"gcp,omitempty"`
+
 	// vsphere is used to configure the vsphere CSI driver.
 	// +optional
 	VSphere *VSphereCSIDriverConfigSpec `json:"vSphere,omitempty"`
 }
 
+// AWSCSIDriverConfigSpec defines properties that can be configured for the AWS CSI driver.
+type AWSCSIDriverConfigSpec struct {
+	// kmsKeyARN sets the cluster default storage class to encrypt volumes with a user-defined KMS key,
+	// rather than the default KMS key used by AWS.
+	// The value may be either the ARN or Alias ARN of a KMS key.
+	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov):kms:[a-z0-9-]+:[0-9]{12}:(key|alias)\/.*$`
+	// +optional
+	KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+}
+
+// AzureDiskEncryptionSet defines the configuration for a disk encryption set.
+type AzureDiskEncryptionSet struct {
+	// subscriptionID defines the Azure subscription that contains the disk encryption set.
+	// The value should meet the following conditions:
+	// 1. It should be a 128-bit number.
+	// 2. It should be 36 characters (32 hexadecimal characters and 4 hyphens) long.
+	// 3. It should be displayed in five groups separated by hyphens (-).
+	// 4. The first group should be 8 characters long.
+	// 5. The second, third, and fourth groups should be 4 characters long.
+	// 6. The fifth group should be 12 characters long.
+	// An Example SubscrionID: f2007bbf-f802-4a47-9336-cf7c6b89b378
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength:=36
+	// +kubebuilder:validation:Pattern:=`^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$`
+	SubscriptionID string `json:"subscriptionID"`
+
+	// resourceGroup defines the Azure resource group that contains the disk encryption set.
+	// The value should consist of only alphanumberic characters,
+	// underscores (_), parentheses, hyphens and periods.
+	// The value should not end in a period and be at most 90 characters in
+	// length.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength:=90
+	// +kubebuilder:validation:Pattern:=`^[\w\.\-\(\)]*[\w\-\(\)]$`
+	ResourceGroup string `json:"resourceGroup"`
+
+	// name is the name of the disk encryption set that will be set on the default storage class.
+	// The value should consist of only alphanumberic characters,
+	// underscores (_), hyphens, and be at most 80 characters in length.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength:=80
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9\_-]+$`
+	Name string `json:"name"`
+}
+
+// AzureCSIDriverConfigSpec defines properties that can be configured for the Azure CSI driver.
+type AzureCSIDriverConfigSpec struct {
+	// diskEncryptionSet sets the cluster default storage class to encrypt volumes with a
+	// customer-managed encryption set, rather than the default platform-managed keys.
+	// +optional
```

Comment:
What happens if this isn't set

Reply:
The comment indicates what happens when the field isn’t set: the default cloud managed key is used.